### PR TITLE
Declare dynamic `Laminas\Crypt\Symmetric\Openssl#$supportedAlgos` property for PHP 8.2 compatibility

### PR DESCRIPTION
**Pull request by basselin**
PHP Deprecated:  Creation of dynamic property

https://github.com/laminas/laminas-crypt/issues/29

|    Q          |   A
|-------------- | ------
| Documentation |no
| Bugfix        | no
| BC Break      | yes
| New Feature   | no
| RFC           | no
| QA            | no

### CODE REVIEW

File: src/Symmetric/Openssl.php [hunk 1]
```diff
@@ -167,6 +167,14 @@ class Openssl implements SymmetricInterface
      */
     protected $tagSize = 16;
 
+    /**
+     * Supported algorithms
+     *
+     * @internal
```
Comment (Ocramius) [inline src/Symmetric/Openssl.php]: Let's please explain why we added this internal annotation, for posterity.

File: src/Symmetric/Openssl.php [hunk 2]
```diff
@@ -167,6 +167,14 @@ class Openssl implements SymmetricInterface
...
+     * @var array
```
Comment (Ocramius) [inline src/Symmetric/Openssl.php]: Can we be more precise with this type? `array<int, string>`? `list<string>`?

**Review by Ocramius — CHANGES_REQUESTED** [review]
Most tests fail due to OpenSSL no longer supporting some operations used in tests: that's fine.

Only one CS issue left then: should be fixable by `./vendor/bin/phpcbf` :)

File: src/Symmetric/Openssl.php (hunk 1 above)
Comment (basselin) [reply src/Symmetric/Openssl.php]: You submitted this suggestion, so I obeyed.

File: src/Symmetric/Openssl.php (hunk 2 above)
Comment (basselin) [reply src/Symmetric/Openssl.php]: I followed the initial code which is not precise on this point.

File: src/Symmetric/Openssl.php (hunk 1 above)
Comment (Ocramius) [reply src/Symmetric/Openssl.php]: > Marking it `@internal` and documenting that it's only there for BC purposes

From https://github.com/laminas/laminas-crypt/issues/29#issuecomment-1446314365

File: src/Symmetric/Openssl.php (hunk 2 above)
Comment (Ocramius) [reply src/Symmetric/Openssl.php]: Yes, and I'm asking for a refinement here, which seems normal? 🤔

Comment (basselin) [reply src/Symmetric/Openssl.php]: It is indeed a `list<string>`.

Comment (Ocramius) [reply src/Symmetric/Openssl.php]: BTW, these adjustments are regular/normal in a PR cycle: you can push fixup commits to the same branch, and this patch will be updated 👍

File: src/Symmetric/Openssl.php (hunk 1 above)
Comment (Ocramius) [reply src/Symmetric/Openssl.php]: CI is still complaining about the incorrect annotations group here: perhaps it wants `@internal` after `@var`, can't remember.

A sentence explaining the reason for `@internal` would be a good idea, for posterity/future maintainers.

Comment (basselin) [reply src/Symmetric/Openssl.php]: What comment do you want for posterity? ^_^"

Comment (Ocramius) [reply src/Symmetric/Openssl.php]: Maybe `@internal this property was declared for compatibility with PHP 8.2, and is not supposed to be used directly, other than for BC reasons` or such?

Comment (Ocramius) [reply src/Symmetric/Openssl.php]: Did you see the CI failures though?

Comment (basselin) [reply src/Symmetric/Openssl.php]: How would you write this comment in two lines? I'm having trouble finding the correct writing convention.
`Incorrect annotations group.`

Comment (Ocramius) [reply src/Symmetric/Openssl.php]: My endorsement would be to try running `./vendor/bin/phpcbf` once, which should attempt to fix it for you.

Comment (basselin) [reply src/Symmetric/Openssl.php]: phpcbf (3.7.2) doesn't seem to be bothered by this comment... despite the 39 errors found...

Comment (Ocramius) [reply src/Symmetric/Openssl.php]: Fair enough: I'll take it to merge from here, will try various things :-)

Thanks meanwhile!

**Review by Ocramius — APPROVED** [review]
Thanks @basselin!

**Pull request merged**